<launch>

    <arg name="rviz" default="true"/>
    <arg name="use_sim_time" default="true"/>
    <arg name="odometry_topic" default="/odom/ground_truth"/>
    <arg name="global_frame" default="odom"/>

    <param name="use_sim_time" value="$(arg use_sim_time)" />

    <!-- Get Pointcloud and transform to Gridmap-->
    <node pkg="detection" type="grid_map_test" name="grid_map_test" output="screen" >
        <param name="point_sub_topic" value="/map_points" />
        <param name="grid_pub_topic" value="/grid_map" />
        <param name="position_sub_topic" value="$(arg odometry_topic)" />
        <param name="grid_frame_id" value="$(arg global_frame)" />
        <param name="point_update" value="true" />
        <param name="position_update" value="true" />
        <param name="grid_map_size_x" value="8.0" />
        <param name="grid_map_size_y" value="8.0" />
        <param name="grid_map_position_x" value="0.0" />
        <param name="grid_map_position_y" value="0.0" />
        <param name="grid_map_resolution" value="0.05" />
    </node>

    <!-- Launch the grid map filters demo node. -->
    <node pkg="grid_map_demos" type="filters_demo" name="grid_map_filter_demo" output="screen">
        <!-- Input topic -->
        <param name="input_topic" value="/grid_map" />
        <!-- Output topic -->
        <param name="output_topic" value="filtered_map" />
        <!-- Load grid map filter chain configuration -->
        <rosparam command="load" file="$(find detection)/config/filters_demo_filter_chain_champ.yaml" />
    </node>

    <node pkg="detection" type="grid_to_occupancy" name="grid_to_occupancy" output="screen">
        <param name="grid_sub_topic" value="/grid_map_filter_demo/filtered_map" />
        <param name="occupancy_grid_pub_topic" value="/occupancy_grid_traversability" />
        <param name="elevation_layer" value="elevation" />
        <param name="traversability_layer" value="traversability" />
        <param name="data_min" value="0.2" />
        <param name="data_max" value="0.0" />
    </node>

    <!-- Load the general parameters (except for filter chain) for this demo. -->
<!--    <rosparam command="load" file="$(find detection)/config/filters_demo.yaml" />-->

    <!-- Launch the grid map visualizer. -->
<!--    <node pkg="grid_map_visualization" type="grid_map_visualization" name="grid_map_visualization" output="screen" />-->

    <!-- Launch RViz with the demo configuration. -->
    <node if="$(arg rviz)" launch-prefix="nice" name="gridmap_rviz" pkg="rviz" type="rviz"
          args="-d $(find detection)/rviz/grid_map_champ.rviz" />

</launch>
